tmt/steps/provision/beaker_mrack.py

```diff
@@ -0,0 +1,462 @@
+import dataclasses
+import datetime
+import sys
+from typing import Any, Dict, List, Optional, Tuple, cast
+
+import click
+
+import tmt
+import tmt.options
+import tmt.steps
+import tmt.steps.provision
+import tmt.utils
+from tmt.utils import ProvisionError, updatable_message
+
+if sys.version_info >= (3, 8):
+    from typing import TypedDict
+else:
+    from typing_extensions import TypedDict
+
+import asyncio
+from functools import wraps
+
+from mrack.providers import providers
+from mrack.providers.beaker import PROVISIONER_KEY as BEAKER
+from mrack.providers.beaker import BeakerProvider
+from mrack.transformers.beaker import BeakerTransformer
+
+import mrack
+
+
+def async_run(func: Any) -> Any:
+    """Decorate click actions to run as async."""
+
+    @wraps(func)
+    def update_wrapper(*args: Any, **kwargs: Any) -> Any:
+        return asyncio.run(func(*args, **kwargs))
+
+    return update_wrapper
+
+
+providers.register(BEAKER, BeakerProvider)
+
+DEFAULT_USER = 'root'
+DEFAULT_ARCH = 'x86_64'
+DEFAULT_PROVISION_TIMEOUT = 3600  # 1 hour timeout at least
+DEFAULT_PROVISION_TICK = 60  # poll job each minute
+
+
+# Type annotation for "data" package describing a guest instance. Passed
+# between load() and save() calls
+GuestInspectType = TypedDict(
+    'GuestInspectType', {
+        "status": str,
+        "system": str,
+        'address': Optional[str]
+        }
+    )
+
+size_translation = {
+    "TB": 1000000,
+    "GB": 1000,
+    "MB": 1,
+    }
+
+
+@dataclasses.dataclass
+class BeakerGuestData(tmt.steps.provision.GuestSshData):
+    # Override parent class with our defaults
+    user: str = DEFAULT_USER
+
+    # Guest request properties
+    arch: str = DEFAULT_ARCH
+    image: Optional[str] = None
+    hardware: Dict[str, Any] = dataclasses.field(default_factory=dict)
+
+    # Provided in Beaker job
+    guestname: Optional[str] = None
+
+    # Timeouts and deadlines
+    provision_timeout: int = DEFAULT_PROVISION_TIMEOUT
+    provision_tick: int = DEFAULT_PROVISION_TICK
+
+
+@dataclasses.dataclass
+class ProvisionBeakerData(BeakerGuestData, tmt.steps.provision.ProvisionStepData):
+    pass
+
+
+GUEST_STATE_COLOR_DEFAULT = 'green'
+
+GUEST_STATE_COLORS = {
+    "New": "blue",
+    "Scheduled": "blue",
+    "Queued": "cyan",
+    "Processed": "cyan",
+    'Waiting': 'magenta',
+    'Installing': 'magenta',
+    "Running": "magenta",
+    "Cancelled": "yellow",
+    "Aborted": "yellow",
+    "Reserved": "green",
+    "Completed": "green",
+    }
+
+
+class BeakerAPI:
+    req: Dict[str, Any] = {}
+    dsp_name: str = "Beaker"
+
+    @async_run
+    async def __init__(self, guest: 'GuestBeaker') -> None:  # type: ignore[misc]
```

Comment:
One note in general: when adding `type: ignore`, please add a short comment explaining why it's needed. I introduced some shared format for these comments, see e.g. https://github.com/teemtee/tmt/blob/main/tmt/utils.py#L2065, https://github.com/teemtee/tmt/blob/main/tmt/utils.py#L2076, https://github.com/teemtee/tmt/blob/main/tmt/base.py#L2748 and so on. If there's an issue or ticket providing more details, mention it, and explain shortly why mypy is wrong/needs a hint/it's safe to ignore its warning.

Reply:
This i need to fix still